flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcDynamicTableSource.java

```diff
@@ -171,17 +205,60 @@ public boolean equals(Object o) {
                 && Objects.equals(lookupOptions, that.lookupOptions)
                 && Objects.equals(physicalRowDataType, that.physicalRowDataType)
                 && Objects.equals(dialectName, that.dialectName)
-                && Objects.equals(limit, that.limit);
+                && Objects.equals(limit, that.limit)
+                && Objects.equals(resolvedPredicates, that.resolvedPredicates);
     }
 
     @Override
     public int hashCode() {
         return Objects.hash(
-                options, readOptions, lookupOptions, physicalRowDataType, dialectName, limit);
+                options,
+                readOptions,
+                lookupOptions,
+                physicalRowDataType,
+                dialectName,
+                limit,
+                resolvedPredicates);
     }
 
     @Override
     public void applyLimit(long limit) {
         this.limit = limit;
     }
+
+    /**
+     * This method makes use of {@link JdbcFilterPushdownVisitor} to generate dialect-specific SQL
+     * expression. The visitor returns Optional.empty() for filter that it cannot handle, which will
+     * then be handled in Flink runtime.
+     */
+    @Override
+    public Result applyFilters(List<ResolvedExpression> filters) {
```

Comment:
Core implementation of this change:

We traverse the ResolvedExpression, and produce a String if we know how to push it to SQL Database, returning None if we cannot handle it.

The unhandled expression will be kept as Flink SQL and run in the job